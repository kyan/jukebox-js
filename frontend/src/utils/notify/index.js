import { store } from 'react-notifications-component'

const Notify = {
  info: ({ id, title, message }) => {
    return store.addNotification({
      id,
      title,
      message,
      type: 'info',
      insert: 'top',
      container: 'bottom-left',
      dismiss: {
        duration: 3000
      }
    })
  },
  success: ({ id, title, message }) => {
    return store.addNotification({
      id,
      title,
      message,
      type: 'success',
      insert: 'top',
      container: 'bottom-left',
      dismiss: {
        duration: 3000
      }
    })
  },
  warning: ({ id, title, message }) => {
    return store.addNotification({
      id,
      title,
      message,
      type: 'warning',
      insert: 'top',
      container: 'bottom-left',
      dismiss: {
        duration: 5000
      }
    })
  }
}

export default Notify
