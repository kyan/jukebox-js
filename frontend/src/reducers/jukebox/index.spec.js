import reducer from './index'
import Types from '../../constants'
import notify from '../../utils/notify'
jest.mock('../../utils/notify')

describe('jukebox', () => {
  it('handles default state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot()
  })

  it('handles a CONNECTED', () => {
    expect(reducer(undefined, {
      type: Types.CONNECTED
    })).toMatchSnapshot()
  })

  it('handles a DISCONNECTED', () => {
    expect(reducer({ online: true }, {
      type: Types.DISCONNECTED
    })).toMatchSnapshot()
  })

  it('handles a UPDATE_VOLUME', () => {
    expect(reducer(undefined, {
      type: Types.UPDATE_VOLUME,
      volume: 32
    })).toMatchSnapshot()
  })

  it('handles a UPDATE_RADIO_STREAM_STATE', () => {
    expect(reducer({ radioStreamPlaying: false }, {
      type: Types.UPDATE_RADIO_STREAM_STATE
    })).toMatchSnapshot()
  })

  it('handles a UPDATE_VOLUME increased', () => {
    const state = {
      volume: 10,
      online: false
    }
    expect(reducer(state, {
      type: Types.UPDATE_VOLUME,
      volume: 32
    })).toMatchSnapshot()
  })

  it('handles a UPDATE_VOLUME decreased', () => {
    const state = {
      volume: 10,
      online: false
    }
    expect(reducer(state, {
      type: Types.UPDATE_VOLUME,
      volume: 5
    })).toMatchSnapshot()
  })

  it('handles a UPDATE_PLAYBACK_STATE playing', () => {
    expect(reducer(undefined, {
      type: Types.UPDATE_PLAYBACK_STATE,
      state: 'playing'
    })).toMatchSnapshot()
    expect(notify.mock.calls.length).toEqual(1)
    expect(notify.mock.calls[0][0]).toEqual('Jukebox playing')
    notify.mockClear()
  })

  it('handles a UPDATE_PLAYBACK_STATE playing when already playing', () => {
    const state = {
      playbackState: 'playing',
      online: false,
      volume: 0
    }
    expect(reducer(state, {
      type: Types.UPDATE_PLAYBACK_STATE,
      state: 'playing'
    })).toMatchSnapshot()
    expect(notify.mock.calls.length).toEqual(0)
    notify.mockClear()
  })
})
