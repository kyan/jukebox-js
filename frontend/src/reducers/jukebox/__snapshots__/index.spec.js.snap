// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jukebox handles a CONNECTED 1`] = `
Object {
  "online": true,
  "playbackState": "paused",
  "radioStreamEnabled": false,
  "radioStreamPlaying": false,
  "volume": 0,
}
`;

exports[`jukebox handles a DISCONNECTED 1`] = `
Object {
  "online": false,
}
`;

exports[`jukebox handles a UPDATE_PLAYBACK_STATE playing 1`] = `
Object {
  "online": false,
  "playbackState": "playing",
  "radioStreamEnabled": false,
  "radioStreamPlaying": false,
  "volume": 0,
}
`;

exports[`jukebox handles a UPDATE_PLAYBACK_STATE playing when already playing 1`] = `
Object {
  "online": false,
  "playbackState": "playing",
  "volume": 0,
}
`;

exports[`jukebox handles a UPDATE_RADIO_STREAM_STATE 1`] = `
Object {
  "radioStreamPlaying": true,
}
`;

exports[`jukebox handles a UPDATE_VOLUME 1`] = `
Object {
  "online": false,
  "playbackState": "paused",
  "radioStreamEnabled": false,
  "radioStreamPlaying": false,
  "volume": 32,
}
`;

exports[`jukebox handles a UPDATE_VOLUME decreased 1`] = `
Object {
  "online": false,
  "volume": 5,
}
`;

exports[`jukebox handles a UPDATE_VOLUME increased 1`] = `
Object {
  "online": false,
  "volume": 32,
}
`;

exports[`jukebox handles default state 1`] = `
Object {
  "online": false,
  "playbackState": "paused",
  "radioStreamEnabled": false,
  "radioStreamPlaying": false,
  "volume": 0,
}
`;
